modules/reporting/lib/widget/filters/user.rb

```diff
@@ -0,0 +1,95 @@
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class Widget::Filters::User < Widget::Filters::Base
+  include AngularHelper
+
+  def render # rubocop:disable Metrics/AbcSize
+    write(content_tag(:div, id: "#{filter_class.underscore_name}_arg_1", class: "advanced-filters--filter-value") do
+      label = html_label
+
+      selected_values = map_filter_values
+
+      box = angular_component_tag "opce-user-autocompleter",
+                                  inputs: {
+                                    InputName: "values[#{filter_class.underscore_name}]",
+                                    hiddenFieldAction: "change->reporting--page#selectValueChanged",
+                                    multiple: true,
+                                    model: selected_values.compact,
+                                    resource: "principals",
+                                    url: ::API::V3::Utilities::PathHelper::ApiV3Path.principals,
+                                    filters: [
+                                      { name: "type", operator: "=", values: ["User"] },
+                                      { name: "status", operator: "!", values: [Principal.statuses["locked"].to_s] }
+                                    ],
+                                    additionalOptions: [
+                                      { id: CostQuery::Filter::UserId.me_value, name: I18n.t(:label_me) }
+                                    ]
+                                  },
+                                  id: "#{filter_class.underscore_name}_select_1",
+                                  class: "filter-value"
+
+      content_tag(:span, class: "inline-label") do
+        label + box
+      end
+    end)
+  end
+
+  private
+
+  def html_label
+    label_tag "#{filter_class.underscore_name}_arg_1_val",
+              "#{h(filter_class.label)} #{I18n.t(:label_filter_value)}",
+              class: "sr-only"
+  end
+
+  def map_filter_values # rubocop:disable Metrics/AbcSize
+    # In case the filter values are all written in a single string (e.g. ["12, 33"])
+    if filter.values.length === 1 && filter.values[0].instance_of?(String)
+      filter.values = filter.values[0].split(",")
+    end
```

Comment:
Since these AbcSize inducing lines come up for the `Project`, the `User` and the `WorkPackage` filter, I think it would be worth it to move this to a helper method of the base class.